code/modules/mob/living/carbon/human/life/handle_regular_hud_updates.dm

```diff
@@ -99,6 +99,15 @@
 		// Rewrite idea : divide life() into organs (Eyes...) and have flags in the roles if they overwrite the functions of those organs.
 		// Basically, the problem here is that abstract things (HUD icons) are handled as the same time as "organs" things (seeing in the dark.)
 		var/datum/role/vampire/V = isvampire(src)
+		if (V)
+			del V.null_rod_wearer
```

Comment:
I did observe a fair bit if this was costly or not. It wasn't on the profiler.
I also have a code for a cached images list on the vamp roles. I do wonder which one performs the best (the difference isn't measurable on the profiler for normal gameplay).